### ISSUE TITLE
"Lambda functions with ECR images" analysis job crashes with closed connection

### ISSUE THREAD

**Issue by achantavy** [lyft/cartography#771]
*Description*:
> What issue is being seen?  Describe what should be happening instead of the bug, for example: Cartography should not crash, the expected value isn't returned, the data schema is wrong, etc.

The [lambda functions with ECR images analysis job](https://github.com/lyft/cartography/blob/master/cartography/data/jobs/analysis/aws_lambda_ecr.json) crashes with this message:

```
Unhandled error while executing statement in job 'Lambda functions with ECR images': Failed to write to closed connection Address(host='HOST', port=7687) (Address(host='IP', port=7687))
```

*To Reproduce*:
> Steps to reproduce the behavior.  Provide all data and inputs required to reproduce the issue.

Run the sync, presumably with a large enough graph

*Logs*:
> If applicable, copy and paste your console log with the failing stack trace.

```
Traceback (most recent call last):
  File "{path}/neo4j/__init__.py", line 682, in _run_transaction
    result = unit_of_work(tx, *args, **kwargs)
  File "{path}/cartography/graph/statement.py", line 92, in _run_noniterative
    summary: neo4j.BoltStatementResultSummary = result.summary()
  File "{path}/neo4j/__init__.py", line 961, in summary
    self.detach()
  File "{path}/neo4j/__init__.py", line 921, in detach
    return self._session.detach(self, sync=sync)
  File "{path}/neo4j/__init__.py", line 563, in detach
    count += fetch()
  File "{path}/neo4j/__init__.py", line 527, in fetch
    detail_count, _ = self._connection.fetch()
  File "{path}/neobolt/direct.py", line 419, in fetch
    return self._fetch()
  File "{path}/neobolt/direct.py", line 440, in _fetch
    self._receive()
  File "{path}/neobolt/direct.py", line 487, in _receive
    raise self.Error(message)
neobolt.exceptions.ServiceUnavailable: Failed to read from defunct connection Address(host='HOST', port=7687) (Address(host='IP', port=7687))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "{path}/neo4j/__init__.py", line 832, in close
    self.sync()
  File "{path}/neo4j/__init__.py", line 797, in sync
    self.session.sync()
  File "{path}/neo4j/__init__.py", line 542, in sync
    detail_count, _ = self._connection.sync()
  File "{path}/neobolt/direct.py", line 527, in sync
    detail_delta, summary_delta = self.fetch()
  File "{path}/neobolt/direct.py", line 419, in fetch
    return self._fetch()
  File "{path}/neobolt/direct.py", line 430, in _fetch
    raise self.Error("Failed to read from closed connection "
neobolt.exceptions.ServiceUnavailable: Failed to read from closed connection Address(host='HOST', port=7687) (Address(host='IP', port=7687))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "{path}/cartography/intel/aws/__init__.py", line 145, in _sync_multiple_accounts
    _sync_one_account(
  File "{path}/cartography/intel/aws/__init__.py", line 80, in _sync_one_account
    run_analysis_job(
  File "{path}/cartography/util.py", line 23, in run_analysis_job
    GraphJob.run_from_json(
  File "{path}/cartography/graph/job.py", line 111, in run_from_json
    job.run(neo4j_session)
  File "{path}/cartography/graph/job.py", line 55, in run
    stm.run(neo4j_session)
  File "{path}/cartography/graph/statement.py", line 71, in run
    session.write_transaction(self._run_noniterative).consume()
  File "{path}/neo4j/__init__.py", line 718, in write_transaction
    return self._run_transaction(WRITE_ACCESS, unit_of_work, *args, **kwargs)
  File "{path}/neo4j/__init__.py", line 708, in _run_transaction
    raise errors[-1]
  File "{path}/neo4j/__init__.py", line 690, in _run_transaction
    tx.close()
  File "{path}/neo4j/__init__.py", line 841, in close
    self.session.rollback_transaction()
  File "{path}/neo4j/__init__.py", line 658, in rollback_transaction
    cx.sync()
  File "{path}/neobolt/direct.py", line 522, in sync
    self.send()
  File "{path}/neobolt/direct.py", line 388, in send
    self._send()
  File "{path}/neobolt/direct.py", line 400, in _send
    raise self.Error("Failed to write to closed connection "
neobolt.exceptions.ServiceUnavailable: Failed to write to closed connection Address(host='HOST', port=7687) (Address(host='IP', port=7687))
```

*Please complete the following information:*:

>  - Cartography release version or commit hash [e.g. 0.12.0 or 95e8e11913e2a44a4d4682506d8364a638ceac69]

0.54.0rc2

**Comment by achantavy** [comment]
Actually, this isn't a code issue - our infra crashed 😅  Closing.

**Comment by achantavy** [comment]
Reopening as I was just able to repro this.

**Comment by ramonpetgrave64** [comment]
Fixed in #816
